string id
float64 grad
float64 hessian
int32 num_links
